src/app/user/update-observation/update-observation.component.ts

```diff
@@ -0,0 +1,92 @@
+import { ObservationService } from './../../services/observation.service';
+import { HttpClient } from '@angular/common/http';
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup } from '@angular/forms';
+import { Router, ActivatedRoute } from '@angular/router';
+import { GeolocationService } from '@ng-web-apis/geolocation';
+import { DatePipe} from '@angular/common';
+
+@Component({
+  selector: 'app-update-observation',
+  templateUrl: './update-observation.component.html',
+  styleUrls: ['./update-observation.component.scss'],
+})
+export class UpdateObservationComponent implements OnInit {
+  public latitude;
```

Comment:
instead of declaring these each as an indivudal class level variable, you could just use your observation variable and then access each property of the observation in the template file - no need to declare this many class level variables.